### ISSUE TITLE
Any way to add/edit custom measures in a PBIX/PBIT for a Live connected AAS connection?

### ISSUE THREAD

**Issue by o-o00o-o** [otykier/TabularEditor#791]
In our PBIX's connected to our AAS database(s), adding/editing custom measures is painfully slow. It reminds me that Microsoft seem to hate Tabular model developers as they force to wait synchronously until they reload the model on every single change to dax, formatting, decimal places etc, First with visual studio and now with Power BI. Either they hate us, or they were on holiday for the responsive asynchronous UI experience Microsoft briefing. 

Anyway, @otykier doesn't hate us, so I thought Tabular Editor could be the solution! :) I know there has been a bunch of things going on with Tabular Editor compatibility with Power BI recently but I wonder if this is a blind-spot as I've tried opening both a pbix but get the error

![image](https://user-images.githubusercontent.com/13693159/110047593-11845500-7d46-11eb-8571-0712dcb205a1.png)

and a pbit but I get the error:

![image](https://user-images.githubusercontent.com/13693159/110047655-2d87f680-7d46-11eb-95e0-9f44d0ba3d9f.png)

This is with a totally clean recent new pbix with nothing in it except the live-connection

I also tried connecting to the local instance server with pbi desktop open but I get this

![image](https://user-images.githubusercontent.com/13693159/110047483-e1d54d00-7d45-11eb-9123-ab4e7e0f299f.png)

Is there no way to do this? We find it useful for prototyping complex measures sometimes, so it would just be a real productivity improvement.

**Comment by otykier** [comment]
I'm sorry to have to disappoint you.

A pbix file that uses a live connection to any instance of AS (SSAS / AAS / PBI datasets) does not in itself contain a tabular model. Hence, it cannot be opened with Tabular Editor. In other words, there is no (supported) way for an external tool to access or modify report level measures defined in a live connected report.

However, if you don't mind using the [new (preview) composite models feature (DirectQuery over AS)](https://powerbi.microsoft.com/en-us/blog/directquery-for-power-bi-datasets-and-azure-analysis-services-preview/), you could convert a report with a live connection, to one that actually contains a model, and THEN it would be possible to edit it using Tabular Editor.

**Comment by o-o00o-o** [comment]
Yes I guessed that might be the answer. I guess it would be possible somehow by manipulating the xml files in the pbix, but imagine that this is unsupported. Still it would be a big advantage.

The problem with DQ models over AS is that they don't work for our model right now

I get 

![image](https://user-images.githubusercontent.com/13693159/110089545-271d6d00-7d8e-11eb-9e41-5f28ae93ae26.png)

and after some time I get

![image](https://user-images.githubusercontent.com/13693159/110089533-2258b900-7d8e-11eb-909d-fa2bc3cc43d7.png)

I haven't had a chance to spend any time on it yet tbh. I've reported it using the button, but that is as far as I've gone

**Comment by otykier** [comment]
Yeah, DirectQuery over AS is still in preview and there are some issues currently. I'm confident the team will have these issues sorted before GA.
